spec/lib/migrations/load_rate_reference_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'rails_helper'
+require 'rake'
+require 'roo'
+
+RSpec.describe 'Load Rating Regions Task', :type => :task do
+  
+  context "rate_reference:update_rating_regions" do
+    before :all do
+      Rake.application.rake_require "tasks/migrations/load_rate_reference"
+      Rake::Task.define_task(:environment)
+    end
+    it 'expect not raise errors' do
+      expect { invoke_task }.not_to raise_error
+    end
+    
+    it 'expect to read xlsx file' do
+      file_path = File.join(Rails.root,'lib', 'xls_templates', "SHOP_ZipCode_CY2017_FINAL.xlsx")
+      xlsx = Roo::Spreadsheet.open(file_path)
+      expect(xlsx.sheet(0).row(2)).to match_array(["01001", "Hampden", "No", "Rating Area 1"])
+      expect(xlsx.sheet(0).row(3)).to match_array(["01002", "Hampshire", "Yes", "Rating Area 1"])
+      expect(xlsx.sheet(0).row(4)).to match_array(["01002", "Franklin", "Yes", "Rating Area 1"])
+      expect(xlsx.sheet(0).row(699)).to match_array(["02791", "Bristol", "No", "Rating Area 6"])
+    end
+    
+    it "should have the right data" do
+      file_path = File.join(Rails.root,'lib', 'xls_templates', "SHOP_ZipCode_CY2017_FINAL.xlsx")
+      xlsx = Roo::Spreadsheet.open(file_path)
+      expect(xlsx.sheet(0).row(2).first.to_i).to eq RateReference.first.zip_code
```

Comment:
This misses a key error because you're implicitly assuming the to_i behavior. 

Check the actual returned values.

```
 context "it creates RateReference elements correctly" do
    let(:rate_reference) { RateReference.first }

      it "assigns the correct zip code" do
        expect(rate_reference.zip_code).to eq('01001')
     end
 end

       expected: "01001"
           got: 1001
     
     (compared using ==)
```